### PR TITLE
feat: Add session query and run session query after change postal code

### PR DESCRIPTION
**Pull request by igorbrasileiro**
## What's the purpose of this pull request?
This PRs adds the session region for VTEX platform and runs the session query when the postal code changes.

## How does it work?
When the user changes the postal code, fetches a new channel that contains regionId as value.

## How to test it?
Set a valid postal code in the postal code and press `Enter`

Also, test the search, add to cart.

## Checklist
- [x] CHANGELOG entry added

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,5 +4,5 @@
   "rules": {
     "react/prop-types": "off"
   },
-  "ignorePatterns": ["static/~partytown/*"]
+  "ignorePatterns": ["static/~partytown/*", "src/server/platforms/vtex.js"]
```
Comment (igorbrasileiro) [inline .eslintrc:7]: The eslint parser is breaking on vtex.js dynamic import.

File: package.json [hunk 2]
```diff
@@ -27,9 +27,10 @@
     "@envelop/graphql-jit": "^1.1.1",
     "@envelop/parser-cache": "^2.2.0",
     "@envelop/validation-cache": "^2.2.0",
-    "@faststore/api": "^1.6.7",
+    "@faststore/api": "https://pkg.csb.dev/vtex/faststore/commit/a83409a6/@faststore/api",
```
Comment (igorbrasileiro) [inline package.json:30]: Remember to fix it.

File: src/server/index.js [hunk 3]
```diff
@@ -43,7 +45,7 @@ const formatError = (err) => {
 const getEnvelop = async () =>
   envelop({
     plugins: [
-      useSchema(await apiSchema),
+      useAsyncSchema(apiSchema),
```
Comment (tlgimenes) [inline src/server/index.js:48]: is it working now? Some time ago I tried using it but it didn't work really well

File: src/components/common/PostalCode/PostalCodeInput.tsx [hunk 4]
```diff
@@ -1,26 +1,62 @@
...
+  RegionQueryQueryVariables,
+} from '@generated/graphql'
 
 import './postal-code-input.scss'
 
 const POSTAL_CODE_STORAGE_KEY = 'main::store::postalCode'
 const POSTAL_CODE_INPUT_ID = 'postal-code-input'
 
+export const RegionQuery = gql`
+  query RegionQuery($session: IStoreSession!) {
```
Comment (tlgimenes) [inline src/components/common/PostalCode/PostalCodeInput.tsx:18]: Region? We shouldn't call it region

File: src/components/common/PostalCode/PostalCodeInput.tsx [hunk 5]
```diff
@@ -1,26 +1,62 @@
...
+  const handleSubmit = async (event: KeyboardEvent<HTMLInputElement>) => {
     const value = ref.current?.value
 
-    if (event.key === 'Enter' && typeof value === 'string') {
-      setPostalCode(value)
+    if (!(event.key === 'Enter' && typeof value === 'string')) {
+      return
     }
+
+    setPostalCode(value)
```
Comment (tlgimenes) [inline src/components/common/PostalCode/PostalCodeInput.tsx:41]: isn't it better to set the postal code after the request was sucessfull?

File: src/components/common/PostalCode/PostalCodeInput.tsx (hunk 4 above)
Comment (tlgimenes) [inline src/components/common/PostalCode/PostalCodeInput.tsx:18]: Shoulnd't this be a mutation?

File: src/sdk/product/useProduct.ts [hunk 6]
```diff
@@ -29,7 +29,7 @@ export const useProduct = <T extends BrowserProductQueryQuery>(
     return {
       locator: [
         { key: 'id', value: productID },
-        { key: 'channel', value: channel },
+        { key: 'channel', value: JSON.parse(channel).salesChannel },
```
[the remark this answers is not in the document]
Comment (igorbrasileiro) [reply src/sdk/product/useProduct.ts:32]: Sure, but the api doesn't handle it, now.

File: src/server/index.js (hunk 3 above)
Comment (igorbrasileiro) [reply src/server/index.js:48]: yeah, it is.

File: src/components/common/PostalCode/PostalCodeInput.tsx (hunk 5 above)
Comment (igorbrasileiro) [reply src/components/common/PostalCode/PostalCodeInput.tsx:41]: Yeah, could be.

File: src/components/common/PostalCode/PostalCodeInput.tsx (hunk 4 above)
Comment (igorbrasileiro) [reply src/components/common/PostalCode/PostalCodeInput.tsx:18]: Yeah, the objective of this query is fetch the region and nothing more.

Comment (igorbrasileiro) [reply src/components/common/PostalCode/PostalCodeInput.tsx:18]: Yeah, sure!

File: CHANGELOG.md [hunk 7]
```diff
@@ -22,7 +22,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `IconSVG` component to load SVG Icons.
 - `Suggestions` component.
 - `SearchHistory` component.
+<<<<<<< HEAD
```
Comment (tlgimenes) [inline CHANGELOG.md:25]: rebase gone wrong

File: gatsby-browser.js [hunk 8]
```diff
@@ -13,6 +13,8 @@ import TestProvider from './src/sdk/tests'
 import { uiActions, uiEffects, uiInitialState } from './src/sdk/ui'
 import storeConfig from './store.config'
 
+const channel = JSON.stringify(storeConfig.channel)
```
Comment (tlgimenes) [inline gatsby-browser.js:16]: stringify on store.config.js so we dont need to do this on both gatsby-ssr and gatsby-browser

File: CHANGELOG.md (hunk 7 above)
Comment (igorbrasileiro) [reply CHANGELOG.md]: Fixed

File: gatsby-browser.js (hunk 8 above)
Comment (igorbrasileiro) [reply gatsby-browser.js]: Fixed here, but parse on server. Ok!?

**Comment by igorbrasileiro** [comment]
@tlgimenes Fixed all comments

**Review by lariciamota — APPROVED** [review]

**Comment by igorbrasileiro** [comment]
@tlgimenes can i merge it?! Is ok for you, right now?

**Review by victorhmp — APPROVED** [review]

**Comment by igorbrasileiro** [comment]
Page result - https://base.vtex.app/
| Metric | Mean | Standard deviation |
|--------|--------|--------|
| Cumulative Layout shift (CLS) | 0.00 | 0.00 |
| First Contentful Paint (FCP) | 1470.92 | 4341.72 |
| First Contentful Paint 3g (FCP) | 3008.65 | 0.30 |
| Largest Contentful Paint (LCP) | 1584.09 | 0.30 |
| Time to Interactive (TTI) | 3360.49 | 10216.52 |
| Total Blocking Time (TBT) | 366.00 | 4563.00 |
| Performance score | 0.903 | 0.000469 |
| JavaScript Execution Time | 1328.84 | 33266.17 |
| Speed Index | 1910.20 | 65.41 |

Page result - https://sfj-1a2dcd6--base.preview.vtex.app/
| Metric | Mean | Standard deviation |
|--------|--------|--------|
| Cumulative Layout shift (CLS) | 0.00 | 0.00 |
| First Contentful Paint (FCP) | 1472.09 | 4168.50 |
| First Contentful Paint 3g (FCP) | 3085.04 | 152172.71 |
| Largest Contentful Paint (LCP) | 1698.43 | 4168.57 |
| Time to Interactive (TTI) | 3156.66 | 900.53 |
| Total Blocking Time (TBT) | 278.92 | 195.63 |
| Performance score | 0.931 | 0.000043 |
| JavaScript Execution Time | 1153.39 | 3390.99 |
| Speed Index | 1544.11 | 3526.08 |

**Pull request merged**